qiita_pet/templates/artifact_ajax/processing_artifact.html

```diff
@@ -0,0 +1,425 @@
+   * the first job in the workflow. It then stores the workflow id in the
+   * "data-wf-id" attribute of the "graph-network-div" div.
+   *
+   */
+  function create_workflow(command_id, params_id, req_params) {
+    $.post('{% raw qiita_config.portal_dir %}/study/process/workflow/', {dflt_params_id: params_id, req_params: JSON.stringify(req_params) })
+      .done(function(data) {
+        if (data.status == 'success') {
+          $("#graph-network-div").attr("data-wf-id", data.workflow_id);
+          add_job_node_to_graph(data.job);
+        }
+        else {
+          bootstrapAlert(data.message, "danger");
+        }
+      });
+  }
+
+  /*
+   * Adds a new job to the workflow
+   *
+   * This function retrieves the information to add a new job to the workflow.
+   * If the workflow still doesn't exist, it calls 'create_workflow'. Otherwise
+   * it calls "create_job".
+   *
+   */
+  function add_job() {
+    var command_id = $("#command-sel").val();
+    var params_id = $("#params-sel").val();
+    // Collect the required parameters
+    var req_params = {}
+    $(".required-parameter").each( function () {
+      req_params[this.id] = this.value;
+    })
+    var workflow_id = $("#graph-network-div").attr("data-wf-id");
+    if (workflow_id == -1) {
+      // This is the first command to be run, so the workflow still doesn't
+      // exist in the system.
+      create_workflow(command_id, params_id, req_params);
+    }
+    else {
+      create_job(command_id, params_id, req_params, workflow_id);
+    }
+    $("#processing-info-div").empty();
+    $('#run-btn').prop('disabled', false);
+    $("#graph-network-div").attr("data-wf-job-count", +$("#graph-network-div").attr("data-wf-job-count") + 1);
+  }
+
+  /*
+   * Load the GUI for a given command parameter
+   *
+   * @param p_name str the name of the parameter
+   * @param param_info object the information of the parameter
+   * @param sel_artifacts_info object with the information of the currently selected artifacts
+   *
+   * This function generates the needed GUI specific to the given parameter type
+   *
+   */
+  function load_parameter_gui(p_name, param_info, sel_artifacts_info) {
+    $('<i>').appendTo("#cmd-opts-div").text(p_name.replace('_', ' ') + ': ');
+    var p_type = param_info[0];
+    var allowed_types = param_info[1];
+    if (p_type == 'artifact') {
+      var sel = $('<select>').appendTo("#cmd-opts-div").attr('id', p_name).attr('name', p_name).addClass('required-parameter');
+      for(var key in sel_artifacts_info) {
+         if(allowed_types.indexOf(sel_artifacts_info[key].type) !== -1) {
+           sel.append($('<option>').attr('value', key).text(sel_artifacts_info[key].name));
+         }
+      }
+    }
+    else if (p_type == 'integer') {
+      var inp = $('<input>').appendTo("#cmd-opts-div").attr('id', p_name).attr('name', p_name).attr('type', 'number').addClass('required-parameter');
+    }
```

Comment:
Oops - I moved this file and started working on this function before our meeting last week. This function is the one that I have to modify in the next functionality, so I would suggest leave as it is since I will be completing this function in my next PR. Does this sound good to you?